windows/table.cpp

```diff
@@ -0,0 +1,275 @@
+{
+	// TODO
+}
+
+void uiTableColumnPartSetTextColor(uiTableColumn *c, int part, int modelColumn)
+{
+	// not implemented
+}
+
+// uiTable implementation
+
+uiWindowsControlAllDefaultsExceptDestroy(uiTable)
+
+uiTable *uiNewTable(uiTableModel *model)
+{
+	uiTable *t;
+	int winStyle = WS_CHILD | LVS_AUTOARRANGE | LVS_REPORT | LVS_OWNERDATA | LVS_SINGLESEL;
+
+	uiWindowsNewControl(uiTable, t);
+	new(&t->columns) std::vector<uiTableColumn*>();		// (initialising in place)
+	t->model = model;
+	t->hwnd = uiWindowsEnsureCreateControlHWND(WS_EX_CLIENTEDGE,
+		WC_LISTVIEW,
+		L"",
+		winStyle,
+		hInstance,
+		NULL,
+		TRUE);
+	model->tables.push_back(t);
+	uiWindowsRegisterWM_NOTIFYHandler(t->hwnd, onWM_NOTIFY, uiControl(t));
+	ListView_SetExtendedListViewStyle(t->hwnd, LVS_EX_FULLROWSELECT | LVS_EX_LABELTIP);
+	// TODO: try LVS_EX_AUTOSIZECOLUMNS
+	int n = (*(model->mh->NumRows))(model->mh, model);
+	ListView_SetItemCountEx(t->hwnd, n, 0);
+	return t;
+}
+
+uiTableColumn *uiTableAppendColumn(uiTable *t, const char *name)
+{
+	uiTableColumn *c = uiprivNew(uiTableColumn);
+	c->name = toUTF16(name);
+	c->t = t;
+	c->modelColumn = -1;	// -1 = unassigned
+	// we defer the actual ListView_InsertColumn call until a part is added...
+	t->columns.push_back(c);
+	return c;
+}
+
+void uiTableSetRowBackgroundColorModelColumn(uiTable *t, int modelColumn)
+{
+	// not implemented
+}
+
+static void uiTableDestroy(uiControl *c)
+{
+	uiTable *t = uiTable(c);
+	uiTableModel *model = t->model;
+	std::vector<uiTable*>::iterator it;
+
+	uiWindowsUnregisterWM_NOTIFYHandler(t->hwnd);
+	uiWindowsEnsureDestroyWindow(t->hwnd);
+	// detach table from model
+	for (it = model->tables.begin(); it != model->tables.end(); ++it) {
+		if (*it == t) {
+			model->tables.erase(it);
+			break;
+		}
+	}
+	// free the columns
+	for (auto col: t->columns) {
+		uiprivFree(col->name);
+		uiprivFree(col);
+	}
+	t->columns.~vector<uiTableColumn*>();	// (created with placement new, so just call dtor directly)
+	uiFreeControl(uiControl(t));
+}
+
+static void uiTableMinimumSize(uiWindowsControl *c, int *width, int *height)
+{
+	uiTable *t = uiTable(c);
+	uiWindowsSizing sizing;
+	int x, y;
+
+	// suggested listview sizing from http://msdn.microsoft.com/en-us/library/windows/desktop/dn742486.aspx#sizingandspacing:
+	// "columns widths that avoid truncated data x an integral number of items"
+	// Don't think that'll cut it when some cells have overlong data (eg
+	// stupidly long URLs). So for now, just hardcode a minimum:
+
+	x = 107;	// in line with other controls
+	y = 14*3;	// header + 2 lines (roughly)
+	uiWindowsGetSizing(t->hwnd, &sizing);
+	uiWindowsSizingDlgUnitsToPixels(&sizing, &x, &y);
+	*width = x;
+	*height = y;
+}
+
+static BOOL onWM_NOTIFY(uiControl *c, HWND hwnd, NMHDR *nm, LRESULT *lResult)
+{
+	uiTable *t = uiTable(c);
+	uiTableModelHandler *mh = t->model->mh;
+	BOOL ret = FALSE;
+
+	switch (nm->code) {
+	case LVN_GETDISPINFO:
+	{
+		NMLVDISPINFO* di = (NMLVDISPINFO*)nm;	

```

Comment:
Style consistency:

```c
		NMLVDISPINFO *di = (NMLVDISPINFO *) nm;	
		LVITEM *item = &(di->item); 
```

Note that the rest of this function isn't consistent either, but we can worry about that later.